src/vue/forms/IssuanceForm.vue

```diff
@@ -0,0 +1,234 @@
+          amountRange: amountRange(
+            0.000001,
+            this.availableAmount ? this.availableAmount.value : 0
+          )
+        },
+        receiver: { required },
+        reference: { required }
+      }
+    }
+  },
+  computed: {
+    assetListValues () {
+      return this.ownedAssets
+        .map(asset => ({
+          value: asset.code,
+          label: `${asset.details.name} (${asset.code})`
+        }))
+    },
+    availableAmount () {
+      if (this.ownedAssets.length && this.form.asset) {
+        const asset = this.ownedAssets
+          .find(asset => asset.code === this.form.asset)
+        return {
+          value: asset.availableForIssuance,
+          currency: asset.code
+        }
+      }
+      return { value: 0 }
+    }
+  },
+  async created () {
+    await this.loadAssets()
+    if (this.ownedAssets.length > 0) {
+      this.form.asset = this.ownedAssets[0].code
+    }
+  },
+  methods: {
+    async submit () {
+      if (!this.isFormValid()) return
+      this.disableForm()
+      try {
+        const receiverBalance =
+          await this.getReceiverBalance(this.form.receiver, this.form.asset)
+
+        if (receiverBalance) {
+          const operation =
+            base.CreateIssuanceRequestBuilder.createIssuanceRequest({
+              asset: this.form.asset,
+              amount: this.form.amount.toString(),
+              receiver: receiverBalance.balanceId,
+              reference: this.form.reference,
+              externalDetails: {}
+            })
+          await Sdk.horizon.transactions.submitOperations(operation)
+          Bus.success('issuance.tokens-issued-msg')
+          this.cancelForm()
+        }
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    async getAccountInfoByEmail (email) {
+      const response = await Sdk.api.users.getPage({ email: email })
+      const account = response.data.find(info => info.email === email)
+      if (account) return account
+      Bus.error('issuance.wrong-email-err')
+      return null
+    },
+    async getReceiverBalance (receiver, asset) {
+      let receiverAccountId
+
+      if (receiver.indexOf('@') !== -1) {
+        const receiverAccount = await this.getAccountInfoByEmail(receiver)
+        if (!receiverAccount) return null
+        receiverAccountId = receiverAccount.id
+      } else {
+        receiverAccountId = receiver
+      }
+
+      try {
+        const response =
+          await Sdk.horizon.account.getBalances(receiverAccountId)
+        const receiverBalance = response.data
+          .find(balance => balance.asset === asset)
+
+        if (receiverBalance) return receiverBalance
+        Bus.error('issuance.balance-required-err')
+        return null
+      } catch (e) {
+        Bus.error('issuance.wrong-account-id-err')
+        console.error(e)
+        return null
+      }
+    },
+    cancelForm () {
+      this.$emit('cancel')
```

Comment:
extract 'cancel' to a local const variable `EVENTS` and use the event name from there. to document event list that the component emits

Reply:
Resolved, updated